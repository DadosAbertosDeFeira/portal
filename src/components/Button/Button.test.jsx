/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Button } from "./index";

describe("Button", () => {
  let user = userEvent.setup();

  beforeEach(() => {
    user = userEvent.setup();
  });

  const makeSut = ({ children = "Click me", ...props }) => {
    return render(<Button {...props}>{children}</Button>);
  };

  it("should call onClick when click in button", async () => {
    const onClick = jest.fn();

    makeSut({ onClick });

    const button = screen.getByRole("button");

    await user.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("renders correctly with default properties", () => {
    const { asFragment } = makeSut({});
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly outline button", () => {
    makeSut({ variant: "outline" });
    const button = screen.getByRole("button");
    expect(button).toMatchSnapshot();
  });
});
