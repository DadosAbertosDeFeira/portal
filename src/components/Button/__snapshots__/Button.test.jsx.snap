// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly outline button 1`] = `
<button
  class="border-primary-dark text-primary-dark font-semibold p-3 border rounded transition ease-in-out"
  type="button"
>
  Click me
</button>
`;

exports[`Button renders correctly with default properties 1`] = `
<button
  class="font-semibold p-3 border rounded transition ease-in-out"
  type="button"
>
  Click me
</button>
`;
